engine-tests/src/api-tests/java/org/terasology/engine/ComponentFieldUriTest.java

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.engine;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import static org.evosuite.runtime.EvoAssertions.*;
+import org.terasology.naming.Name;
+import org.terasology.testUtil.ApiTestsParams;
+import org.terasology.testUtil.ApiTestsUtils;
+import org.terasology.testUtil.MethodSignature;
+
+import java.lang.reflect.Modifier;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ComponentFieldUriTest {
+
+    private static Class<ComponentFieldUri> cl;
+
+    private static Set<MethodSignature> registeredMethods;
+
+    private static Set<MethodSignature> registeredConstructors;
+
+    @BeforeClass
+    public static void setupClass() {
+        cl = ComponentFieldUri.class;
+        registeredMethods = ApiTestsUtils.getModifiableSet(ApiTestsParams.ObjectClassMethods);
+        registeredMethods.add(new MethodSignature("getObjectName", null,Modifier.PUBLIC,Name.class,null));
+        registeredMethods.add(new MethodSignature("getModuleName",null,Modifier.PUBLIC,Name.class,null));
+        registeredMethods.add(new MethodSignature("getComponentUri",null,Modifier.PUBLIC,SimpleUri.class,null));
+        registeredMethods.add(new MethodSignature("getFieldName", null,Modifier.PUBLIC,String.class,null));
+        registeredMethods.add(new MethodSignature("isValid",null,Modifier.PUBLIC,boolean.class,null));
+
+        registeredConstructors = new HashSet<>();
+        registeredConstructors.add(new MethodSignature("ComponentFieldUri",new Class[]{SimpleUri.class,String.class},Modifier.PUBLIC,null));
+        registeredConstructors.add(new MethodSignature("ComponentFieldUri",new Class[]{String.class},Modifier.PUBLIC,null));
+    }
+
+    @Test
+    public void majorIncreaseTest() {
+        ApiTestsUtils.compareConstructors(cl,registeredConstructors);
+        ApiTestsUtils.compareMethods(cl,registeredMethods);
+    }
+
+    @Test
+    public void minorIncreaseTest() {
+        ApiTestsUtils.compareConstructorsNum(cl,registeredConstructors);
+        ApiTestsUtils.compareMethodsNum(cl,registeredMethods);
+
+    }
+
+    @Test(timeout = 4000)
+    public void constructorTest1()  throws Throwable  {
```

Comment:
@Cervator If in a pr someone changes the code in a method, then the method returns a different value. In this case, some auto-generated test will fail. It needs a major increase or a minor increase? IMHO it's a major increase.

Reply:
Tricky situation. Based solely on SemVer I don't think its spec cares one way or another: so long as the signatures match you pass SemVer. Logic changes are assumed to be benevolent (working toward fixing bugs or providing new features). A failed unit test either means you introduced a bug or failed to update a unit test, which should block a PR for a reason unrelated to SemVer.

So: broken unit test = fix your code! No merge for you until then :-) SemVer doesn't really matter until the PR qualifies as merge-able.